src/pudl/transform/ferc1.py

```diff
@@ -32,9 +33,141 @@
+            "MaintenanceOfStructuresSteamPowerGeneration": "opex_structures",
+            "MaintenanceOfBoilerPlantSteamPowerGeneration": "opex_boiler",
+            "MaintenanceOfElectricPlantSteamPowerGeneration": "opex_plants",
+            "MaintenanceOfMiscellaneousSteamPlant": "opex_misc_steam",
+            "PowerProductionExpensesSteamPower": "opex_production_total",
+            "ExpensesPerNetKilowattHour": "opex_per_kwh",
+        },
+    },
+    "fuel_ferc1": {
+        "dbf": {
+            "respondent_id": "utility_id_ferc1",
+            "plant_name": "plant_name_ferc1",
+            "fuel": "fuel_type_code_pudl",
+            "fuel_unit": "fuel_units",
+            "fuel_avg_heat": "fuel_btu_per_unit",
+            "fuel_quantity": "fuel_consumed_units",
+            "fuel_cost_burned": "fuel_cost_per_unit_burned",
+            "fuel_cost_delvd": "fuel_cost_per_unit_delivered",
+            "fuel_cost_btu": "fuel_cost_per_mmbtu",
+            "fuel_generaton": "fuel_mmbtu_per_kwh",
+        },
+        "xbrl": {
+            "PlantNameAxis": "plant_name_ferc1",
+            "FuelKindAxis": "fuel_type_code_pudl",
+            "FuelUnit": "fuel_units",
+            "FuelBurnedAverageHeatContent": "fuel_mmbtu_per_unit",
+            "QuantityOfFuelBurned": "fuel_consumed_units",
+            "AverageCostOfFuelPerUnitBurned": "fuel_cost_per_unit_burned",
+            "AverageCostOfFuelPerUnitAsDelivered": "fuel_cost_per_unit_delivered",
+            "AverageCostOfFuelBurnedPerMillionBritishThermalUnit": "fuel_cost_per_mmbtu",
+            "AverageCostOfFuelBurnedPerKilowattHourNetGeneration": "fuel_cost_per_kwh",
+            "AverageBritishThermalUnitPerKilowattHourNetGeneration": "fuel_mmbtu_per_kwh",
+        },
+    },
+}
+
+TABLE_AXIS_COLS_XBRL = {
+    table_name: [  # pudl column names if "Axis" is at the end of the xbrl name
+        xbrl_col for xbrl_col in rename_dicts["xbrl"].keys() if "Axis" in xbrl_col
+    ]
+    for (table_name, rename_dicts) in COLUMN_RENAME.items()
+}
+
+TABLE_PKS_XBRL: dict[TABLES_LITERAL, list] = {
+    table_name: ["report_year", "entity_id"]  # other primary keys
+    + [COLUMN_RENAME[table_name]["xbrl"][axis_col] for axis_col in axis_cols]
+    for (table_name, axis_cols) in TABLE_AXIS_COLS_XBRL.items()
+}
+"""A dictionary of table names (keys) to list of primary keys (values). This is
+derived from the :py:const:`COLUMN_RENAME` dictionary by grabbing the list of pudl
+column names when the xbrl column name has 'Axis' at the end. XBRL uses 'Axis'
+as an indicator for primary keys. We also add 'entity_id' and 'report_date' to
+the primary keys."""
+
 ##############################################################################
 # Dicts for categorizing freeform strings ####################################
 ##############################################################################
+
 FUEL_STRINGS: dict[str, list[str]] = {
```

Comment:
`pudl.helpers.cleanstrings()` provides the same functionality as the `pudl.metadata.Encoder` right? If use the Encoder class, `FUEL_STRINGS` can live in `pudl.metadata.codes`